### DISCUSSION TITLE
possible to have standalone version

### DISCUSSION THREAD

**Discussion by d0uub** [Budibase/budibase#1699] [Ideas]
our environment is strict to security and new technology and docker is not allowed. it cannot access to intranet data if deploy to public cloud. Possible to have standalone exe version?

**Comment by slfhstr** [comment]
How is that a bug ?!

**Comment by mike12345567** [comment]
Hi @d0uub - we've moved this to a discussion as this is an idea rather than an issue with the system.

The way that Budibase is built and used it would make it very very difficult to build a single executable from which the whole system could run, as this would need to include CouchDB, Redis, multiple node services, MinIO/file storage systems - this also wouldn't scale very well to larger deployments - I'm not sure this is very feasible for Budibase.

**Comment by mjashanks** [comment]
@d0uub are you able to have a call to talk more about your setup? https://calendly.com/mjashanks/budibase

Or, feel free to email me - my address is in my profile.

**Comment by slfhstr** [comment]
Given the comments at various times and in various places about focus on further development, I would be highly disappointed if Budibase team spent any more than 5 seconds on this.  Close it and ignore it.  Fringe case if I ever saw one.

**Reply by mjashanks** [reply]
I'm interested because self hosting in a high security network is a great use of budibase. Like @mike12345567 says, packaging into a single `.exe` is out of scope, but it might not be the only option.

Plus - I'm not on the dev team - it's my job to chase these things down, even for future consideration :)

Tim - please book a call with me, or dm me on Discord - we haven't gone away - I'm here to listen.

**Reply by slfhstr** [reply]
If they don't allow docker, they're not going to allow an .exe from a new developer w/o access to source code, code audits etc.
If they are using Windows, then they're not a high security environment.
Happy to chat sometime but bit chocka today.

**Comment by d0uub** [comment]
i saw that there have standalone before (https://github.com/Budibase/budibase/releases/download/v0.​8.13/Budibase-Builder-Setup-0.8.13.exe)
we can pass it to administrator for vulnerability scan for exe and execution in windows server somehow (it still need to access internet for download data as i tried)

we cannot run docker since:
1. linux env only allow to run java related war / ear project
2. windows cannot install docker setup by restricted windows server governance by network team 
3. production server have no internet access 
4. public cloud cannot access internal network data

different company have unique env and restriction based on company culture, business use case especially enterprise level infrastructure we have to stick with those rules. any assumption of exe and docker use case is too narrow in our case.

high security network, strict env and traditional technology is very common in large enterprise company. @mjashanks, just see how you identify your product position

thank you to spend for 5 second about this issue since i am not often use github create issue and i should use blank one

**Reply by mjashanks** [reply]
Thanks for taking the time to add this detail.

The version of Budibase that you refer to was only standalone for a **development** environment. If you were going to deploy and run those apps within your organisation, you would still have to deploy a production deployment using docker - this is what we offered for that version.

To help you understand why - Budibase is composed of several components. 
- Budibase web server - a NodeJS process that hosts your apps and their API 
- Budibase worker - a NodeJS process that runs background tasks - i.e. automations
- CouchDB - a full database that stores Budibase's internal tables and Budibase's app metadata
- Redis - a key/value database - caching layer
- MinIO - a file storage system
- Envoy - a proxy

Each one of these components is a separate piece of software - thus runs in its own docker container. When running larger workloads in production, you would likely want some or all of these to be running in their own VM. 

It's really not feasible (or sensible) to package these all up into one executable. So that will not happen. I'm still happy to discuss other options with you, but it sounds like Budibase might not be a fit for you.

**Reply by d0uub** [reply]
thanks and it is clear

**Comment by TheArthurAbbott** [comment]
I know of a company that does not allow containerized envs run on-site and will only support the cloud. However, the user requirements forbid individual logins (to limit production down password resets, among other workflow issues) whereas their internal IT policy forces them to use MFA for cloud apps.

A client-only installation with central SQL server access, not necessarily a standalone exe, would work. I do think this may be an edge case, but telling someone to spend no more than 5 seconds on this is just plain rude and also irresponsible. It's quite a comparable feature to have an exportable application in comparison to other tools on the market like Plasmic or Builder.io and the like. Perks of centralization like automatic updates can also be deployed in tandem - checking a central location like a network folder for json content, for instance.
